test/integration/pack/test_pack.py

```diff
@@ -1732,9 +1733,268 @@ def test_rocks_noncache_flag(cartridge_cmd, light_project, tmpdir, pack_format):
+
+    if os.path.exists(get_rocks_cache_path()):
+        shutil.rmtree(get_rocks_cache_path())
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        project.path
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+    assert "Using cached path" not in output
+    assert f"File {os.path.join(project.path, 'pack-cache.yml')} with pack cache parameters doesn't exists"
+    assert len(os.listdir(get_rocks_cache_path())) == 0
+
+
+@pytest.mark.parametrize('pack_format', ['tgz'])
+def test_always_cache_path(cartridge_cmd, light_project, tmpdir, pack_format):
+    project = light_project
+    project_dir = hashlib.sha1(project.path.encode('utf-8')).hexdigest()[:10]
+
+    new_cache_yml_path = os.path.join(tmpdir, "new-pack-cache.yml")
+    with open(new_cache_yml_path, "w") as f:
+        yaml.dump({".rocks": {"always-cache": True}}, f)
+
+    replace_project_file(project, "pack-cache.yml", new_cache_yml_path)
+    if os.path.exists(get_rocks_cache_path()):
+        shutil.rmtree(get_rocks_cache_path())
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        project.path
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+    assert "Using cached path .rocks" not in output
+
+    project_path_cache = os.path.join(get_rocks_cache_path(), project_dir, ".rocks")
+    cache_dir_items = os.listdir(project_path_cache)
+    assert len(cache_dir_items) == 1
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+    assert "Using cached path .rocks" in output
+
+    project_path_cache = os.path.join(get_rocks_cache_path(), project_dir, ".rocks")
+    new_cache_dir_items = os.listdir(project_path_cache)
+    assert len(new_cache_dir_items) == 1
+    assert cache_dir_items == new_cache_dir_items
+
+
+@pytest.mark.parametrize('pack_format', ['tgz'])
+def test_multiple_paths_in_pack_file(cartridge_cmd, light_project, tmpdir, pack_format):
+    project = light_project
+    project_dir = hashlib.sha1(project.path.encode('utf-8')).hexdigest()[:10]
+    project_path_cache = os.path.join(get_rocks_cache_path(), project_dir)
+
+    first_cached_dir_path = os.path.join(project.path, "first-dir")
+    second_cached_dir_path = os.path.join(project.path, "second-dir", "nested")
+    os.mkdir(first_cached_dir_path)
+    os.makedirs(second_cached_dir_path)
+
+    with open(os.path.join(first_cached_dir_path, "fst.txt"), "w") as f:
+        f.write("Dummy text 123")
+
+    with open(os.path.join(second_cached_dir_path, "snd.txt"), "w") as f:
+        f.write("321 txet ymmuD")
+
+    new_cache_yml_path = os.path.join(tmpdir, "new-pack-cache.yml")
+    with open(new_cache_yml_path, "w") as f:
+        yaml.dump({
+            ".rocks": {"always-cache": True},
```

Comment:
Why? At ``instances.yml`` and ``replicasets.yml`` we use the same scheme: 
* for ``instances.ym``l key is instance name;
* for ``replicasets.yml`` key also is instance.

And we don't use ``instance-name`` or something like this field.

Reply:
Paths like `.rocks`  contain special characters and should be quoted. It looks ugly for map keys. Github action [`cache`](https://github.com/actions/cache) uses `path` key for that purpose. It can be more convenient for users who configure GitHub CI. I suggest just thinking about it.

Reply:
Agree